app/js/Tab.js

```diff
@@ -65,10 +67,10 @@ class Tab {
    * @param {Integer} tabId
    */
 
-  static generateTabHeaderTemplate(tabId) {
+  static generateTabHeaderTemplate(tabId, tabName) {
     return `
       <li data-tab="${tabId}">
-        <a href="#tab-${tabId}">&nbsp;</a>
+        <a href="#tab-${tabId}">${tabName || "Untitled-" + tabId }</a>
```

Comment:
why tabId ? The point is to put the filename, if the filename isn't provided then we fall back to the Tab id? should it be ${tabName || tabId + " [no name]"}